### PR TITLE
Fix getargs argument passing

### PR DESCRIPTION
**Pull request by hauntsaninja**
Fixes https://github.com/mypyc/mypyc/issues/1078

Introduced in https://github.com/python/mypy/pull/17930

See the first commit to see the bug (wrong condition)

### CODE REVIEW

File: mypyc/lib-rt/getargs.c [hunk 1]
```diff
@@ -374,8 +374,6 @@ vgetargskeywords(PyObject *args, PyObject *kwargs, const char *format,
             int res = PyDict_GetItemStringRef(kwargs, kwlist[i], &current_arg);
             if (res == 1) {
```
Comment (cdce8p) [inline mypyc/lib-rt/getargs.c:375]:
```suggestion
            if (unlikely(res == 1)) {
```
To fully restore the behavior prior to #17930.

**Review by cdce8p — CHANGES_REQUESTED** [review]
Thanks for bisecting the issue @hauntsaninja! Seems I missed that in #17930. Left one small comment, LGTM otherwise.

File: mypyc/test-data/run-classes.test [hunk 2]
```diff
@@ -467,6 +467,14 @@ a = A(10)
 assert a.foo() == 11
 assert foo() == 21
 
+[case testClassKwargs]
+class X:
+    def __init__(self, msg: str, **variables: int) -> None:
+         pass
+[file driver.py]
+from native import X
+X('hello', a=0)
```
Comment (cdce8p) [inline mypyc/test-data/run-classes.test:476]:
```suggestion
import traceback
from native import X
X('hello', a=0)
try:
    X('hello', msg='hello')
except TypeError:
    traceback.print_exc()
[out]
Traceback (most recent call last):
  File "driver.py", line 5, in <module>
    X('hello', msg='hello')
    ~^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument for __init__() given by name ('msg') and position (1)
```
We could also add an additional example to test the error case.

**Comment by hauntsaninja** [comment]
Thanks for the review! I also looked over the CPython getargs code and made some cosmetic changes to make this look more like that

File: mypyc/test-data/run-classes.test [hunk 3]
```diff
@@ -467,6 +467,21 @@ a = A(10)
...
+         pass
+[file driver.py]
+import traceback
+from native import X
+X('hello', a=0)
+try:
+    X('hello', msg='hello')
+except TypeError as e:
+    print(f"{type(e).__name__}: {e}")
+[out]
```
Comment (cdce8p) [inline mypyc/test-data/run-classes.test:482]:
```suggestion
    traceback.print_exc()
[out]
Traceback (most recent call last):
  File "driver.py", line 5, in <module>
    X('hello', msg='hello')
[out version>=3.11]
Traceback (most recent call last):
  File "driver.py", line 5, in <module>
    X('hello', msg='hello')
    ~^^^^^^^^^^^^^^^^^^^^^^
```

The other tests all use conditional `out`. _Not sure which version added the `^^^^` here though, could be `3.11` or maybe `3.13`._

**Pull request merged**